Readme.md

```diff
@@ -18,171 +18,444 @@ This is an ipm package, which contains one or more reusable assets within the ip
+<dd><p>Updates i.e. Replaces the current dataset with the passed dataset</p>
+</dd>
+</dl>
+
+<a name="BigQuery"></a>
+
+## BigQuery(options) ⇒ <code>Object</code>
 BigQuery object requires an API Key
 
 **Kind**: global function  
+**Returns**: <code>Object</code> - Project  
 
 | Param | Type | Description |
 | --- | --- | --- |
-| auth_key | <code>string</code> | is the API key for Google BigQuery Service, https://cloud.google.com/bigquery/docs/authorization |
-
+| options | <code>Object</code> | {      authToken: YOUR_BIGQUERY_AUTH_TOKEN (is the API key for Google BigQuery Service, https://cloud.google.com/bigquery/docs/authorization)      (optional parameters)      ,projectID: YOUR_PROJECT_ID      ,datasetID: DATASET_IN_ABOVE_PROJECT      ,tableID: TABLE_IN_ABOVE_DATASET  } |
 
-* [BigQuery(auth_key)](#BigQuery)
-    * [.delete(projectID, datasetID, callback)](#BigQuery+delete)
-    * [.get(projectID, datasetID, callback)](#BigQuery+get)
-    * [.insert(projectID, requestBody, callback)](#BigQuery+insert)
-    * [.insertAll(projectID, datasetID, tableID, requestBody, callback)](#BigQuery+insertAll)
-    * [.list(projectID, callback)](#BigQuery+list)
-    * [.update(projectID, datasetID, requestBody, callback)](#BigQuery+update)
+**Example**  
+```js
+:- Inside of a ClearBlade service
 
-<a name="BigQuery+delete"></a>
+var project = BigQuery(options).initialize();
 
-### bigQuery.delete(projectID, datasetID, callback)
-Datasets - Delete
+Tip: If user wants to work with 
+1. Another dataset(assuming it exists) in a project then it can be done by:
+     var anotherDS = project.Dataset('YOUR_DATASETID'); // returns a Dataset Object !
+2. Another table(assuming it exists) in a dataset then it can be done by:
+     var anotherTable = project.Dataset.Table('YOUR_TABLEID');
+```
 
-Deletes the dataset specified by the datasetId value. Before you can delete a dataset, 
-you must delete all its tables, either manually or by specifying deleteContents. 
-Immediately after deletion, you can create another dataset with the same name.
+* [BigQuery(options)](#BigQuery) ⇒ <code>Object</code>
+    * [~Project(projectID)](#BigQuery..Project) ⇒ <code>Object</code>
+        * [~Dataset(datasetID)](#BigQuery..Project..Dataset) ⇒ <code>Object</code>
+            * [~Table(tableID)](#BigQuery..Project..Dataset..Table) ⇒ <code>Object</code>
+                * [~InsertAll(requestBody, callback)](#BigQuery..Project..Dataset..Table..InsertAll)
+            * [~Get(callback)](#BigQuery..Project..Dataset..Get) ⇒
+            * [~Delete(callback)](#BigQuery..Project..Dataset..Delete)
+            * [~Insert(requestBody, callback)](#BigQuery..Project..Dataset..Insert) ⇒
+            * [~Update(requestBody, callback)](#BigQuery..Project..Dataset..Update)
+            * [~List(callback)](#BigQuery..Project..Dataset..List)
+        * [~List(callback)](#BigQuery..Project..List)
+
+<a name="BigQuery..Project"></a>
+
+### BigQuery~Project(projectID) ⇒ <code>Object</code>
```

Comment:
After defining the @typedef Project, this function should then return a Project object

Reply:
Removed project object, issue closed